scripts/fdbshow

```diff
@@ -0,0 +1,192 @@
+            ent = self.db.get_all('ASIC_DB', s, blocking=True)
+            port_id = ent[b"SAI_BRIDGE_PORT_ATTR_PORT_ID"][oid_pfx:]
+            self.if_br_oid_map[br_port_id] = port_id 
+
+
+    def fetch_fdb_data(self):
+        """
+            Fetch FDB entries from ASIC DB. 
+            FDB entries are sorted on "VlanID" and stored as a list of tuples
+        """
+        self.db.connect(self.db.ASIC_DB)
+        self.bridge_mac_list = []
+        
+        fdb_str = self.db.keys('ASIC_DB', "ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY:*")
+        if not fdb_str:
+            return
+
+        oid_pfx = len("oid:0x")
+        for s in fdb_str:
+            fdb_entry = s.decode()
+            try:
+                fdb = json.loads(fdb_entry .split(":", 2)[-1])
+            except ValueError as e: 
+                print e.message
+                continue
+           
+            ent = self.db.get_all('ASIC_DB', s, blocking=True)
+            br_port_id = ent[b"SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID"][oid_pfx:]
+            try:
+                port_id = self.if_br_oid_map[br_port_id]
+                if_name = self.if_oid_map[port_id]
+            except KeyError as e:
+                continue
+            
+            self.bridge_mac_list.append((int(fdb["vlan"]),) + (fdb["mac"],) + (if_name,))
+
+        self.bridge_mac_list.sort(key = lambda x: x[0])
+        return
+    
+    
+    def get_iter_index(self, key_value=0, pos=0):
+        """
+            Get the starting index of matched entry
+        """
+        if pos == 0:
+            self.bridge_mac_list.sort(key = lambda x: x[pos])
+        keys = [r[pos] for r in self.bridge_mac_list]
+        try:
+            return keys.index(key_value)
+        except IndexError as e:
+            print e.message
+
+
+    def display_specific(self, port, vlan):
+        """
+            Display the FDB entries for specified vlan/port.
+        """
+        output = []
+
+        if vlan is not None and port is None:
+            vlan = int(vlan)
+            s_index = self.get_iter_index(vlan)
+            fdb_list = [fdb for fdb in self.bridge_mac_list[s_index:] if fdb[0] == vlan]
+
+        elif vlan is None and port is not None:
+            self.bridge_mac_list = natsorted(self.bridge_mac_list, key = lambda x: x[2])
+            s_index = self.get_iter_index(port, 2)
+            fdb_list = [fdb for fdb in self.bridge_mac_list[s_index:] if fdb[2] == port]
+
+        else:
+            vlan = int(vlan)
+            s_index = self.get_iter_index(vlan)
+            fdb_list = [fdb for fdb in self.bridge_mac_list[s_index:]
+                        if fdb[0] == vlan and fdb[2] == port]
+
+        for fdb in fdb_list:
+            self.FDB_COUNT += 1
+            output.append([self.FDB_COUNT, fdb[0], fdb[1], fdb[2]])
+
+        print tabulate(output, self.HEADER)
+        print "Total number of entries {0} ".format(self.FDB_COUNT)
+
+
+    def display_summary(self):
```

Comment:
>display_summary [](start = 8, length = 15)

I guess you don't need one more function for no filters. Just reuse display_specific().

Reply:
Addressed